internal/log/log.go

```diff
@@ -117,36 +117,55 @@ func PrepareLogging(fs afero.Fs, verbose *bool, loggerSpy io.Writer) {
 		loglevel = loggers.LevelDebug
 	}
 
-	logFile, err := prepareLogFile(fs)
+	logFile, errFile, err := prepareLogFiles(fs)
 	logFormat := loggers.ParseLogFormat(os.Getenv(loggers.EnvVarLogFormat))
 	logTime := loggers.ParseLogTimeMode(os.Getenv(loggers.EnvVarLogTime))
 
 	setDefaultLogger(loggers.LogOptions{
-		File:               logFile,
-		FileLoggingJSON:    logFormat == loggers.LogFormatJSON,
-		ConsoleLoggingJSON: logFormat == loggers.LogFormatJSON,
-		LogLevel:           loglevel,
-		LogSpy:             loggerSpy,
-		LogTimeMode:        logTime,
+		File:        logFile,
+		ErrorFile:   errFile,
+		JSONLogging: logFormat == loggers.LogFormatJSON,
+		LogLevel:    loglevel,
+		LogSpy:      loggerSpy,
+		LogTimeMode: logTime,
 	})
 
 	if err != nil {
 		Warn(err.Error())
 	}
 }
 
-func prepareLogFile(fs afero.Fs) (afero.File, error) {
-	timestamp := timeutils.TimeAnchor().Format(TrafficLogFilePrefixFormat)
+// LogFilePath returns the path of a logfile for the current execution time - depending on when this function is called such a file may not yet exist
+func LogFilePath() string {
+	timestamp := timeutils.TimeAnchor().Format(LogFileTimestampPrefixFormat)
+	return filepath.Join(LogDirectory, timestamp+".log")
+}
+
+// ErrorFilePath returns the path of an error logfile for the current execution time - depending on when this function is called such a file may not yet exist
+func ErrorFilePath() string {
+	timestamp := timeutils.TimeAnchor().Format(LogFileTimestampPrefixFormat)
+	return filepath.Join(LogDirectory, timestamp+"-errors.log")
+}
+
+func prepareLogFiles(fs afero.Fs) (logFile afero.File, errFile afero.File, err error) {
 	if err := fs.MkdirAll(LogDirectory, 0777); err != nil {
-		return nil, fmt.Errorf("unable to prepare log directory %s: %w", LogDirectory, err)
+		return nil, nil, fmt.Errorf("unable to prepare log directory %s: %w", LogDirectory, err)
 
 	}
-	logFilePath := filepath.Join(LogDirectory, timestamp+".log")
-	logFile, err := fs.OpenFile(logFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+
+	logFilePath := LogFilePath()
+	logFile, err = fs.OpenFile(logFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
 	if err != nil {
-		return nil, fmt.Errorf("unable to prepare log file in %s directory: %w", LogDirectory, err)
+		return nil, nil, fmt.Errorf("unable to prepare log file in %s directory: %w", LogDirectory, err)
 	}
-	return logFile, nil
+
+	errFilePath := ErrorFilePath()
+	errFile, err = fs.OpenFile(errFilePath, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to prepare error file in %s directory: %w", LogDirectory, err)
```

Comment:
So logFile is opened, but if this errors, it will stay open even if it is never used. Could just close it here as well.